controlplane/kubeadm/controllers/controller.go

```diff
@@ -413,6 +428,41 @@ func (r *KubeadmControlPlaneReconciler) reconcileHealth(ctx context.Context, clu
 	return nil
 }
 
+// ReconcileInfrastructureMachineRefs adds infrastructure template ref as annotation to the infrastructure objects missing it.
+func (r *KubeadmControlPlaneReconciler) ReconcileInfrastructureMachineRefs(ctx context.Context, infraTemplateRef corev1.ObjectReference, machines internal.FilterableMachineCollection) (bool, error) {
+	isPatched := false
+	for _, machine := range machines {
+
+		infraObj, err := external.Get(ctx, r.Client, &machine.Spec.InfrastructureRef, machine.Spec.InfrastructureRef.Namespace)
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				return isPatched, errors.Wrapf(err, "Machine's infrastructure object is missing.")
```

Comment:
```suggestion
				return isPatched, errors.Wrapf(err, "Machine's infrastructure object is missing")
```

Should we continue instead of returning an error in this case?

Reply:
If we are okay with continuing with machines that don't have annotations during reconcile, yes. I think this should not be best effort.

Reply:
I was thinking about machines about to be deleted, which is where this would fail

Reply:
How about returning error only if the machine does not have a deletion timestamp? If we don't return errors at all, reconcile will continue.

Reply:
Yeah, at the beginning of the for loop we can `continue` if we have a machine with a deletion timestamp. In addition, to make sure we're not running this backfill on newly created machines, should we check that `status.infrastructureReady` is true?

Reply:
Good point. 
In case of adopted machines that is not ready, if we continue before `status.infrastructureReady= true` for all machines, it will unnecessarily move down the code without having an annotation and probably stop at reconcilehealth().

Or we can wait until all machines' `status.infrastructureReady=true`. In this case, we need to make sure machine remediation is happening before this.

Reply:
Wouldn't an adopted machine be ready?